easysocial/easysocial/comments.php

```diff
@@ -62,8 +62,16 @@ public function post()
 			$valid				=	0;
 		}
 
-		// Message should not be empty.
-		if (empty($input))
+		// Determine if this user has the permissions to add comment.
+		$access 	= FD::access();
+		$allowed	= $access->get('comments.add');
+
+		if (!$allowed)
+		{
+			$res->result->message = JText::_('PLG_API_EASYSOCIAL_COMMENT_NOT_ALLOW_MESSAGE');
+			$res->result->status = false;
```

Comment:
@Pratikshakhandagale  - Use ApiError::raiseError method with proper status code (403) for unauthorised call.